Codes/SlippiReplays/Record.cpp

```diff
@@ -23,164 +23,90 @@ namespace ReplaysLogic {
+                        startReplay->players[i].startPlayer.yPos = yPos;
+                        startReplay->players[i].startPlayer.zPos = zPos;
+                    }
+
+                    EXIPacket startReplayPacket = EXIPacket(EXIStatus::START_REPLAYS_STRUCT, startReplay, sizeof(StartReplay));
+                    startReplayPacket.Send();
+                    delete startReplay;
                 }
-            }
-            ftManager *fighter = FIGHTER_MANAGER;
-            const GameGlobal *game = GAME_GLOBAL;
-            itemIdName *itemIds;
-            u16 *itemVariants;
-            u32 *gameInfo = new u32[2];
-            int sizeOfItems = ITEM_MANAGER->baseItemArrayList.size();
-
-            gameInfo[0] = game->gameFrame->frameCounter;
-            gameInfo[1] = game->gameFrame->persistentFrameCounter;
-
-            EXIPacket gamePacket = EXIPacket(EXIStatus::GAME, gameInfo, 2 * sizeof(u32));
-            gamePacket.Send();
-            delete[] gameInfo;
-
-            if(sizeOfItems > 0) {
-                itemIds = new itemIdName[sizeOfItems];
-                itemVariants = new u16[sizeOfItems];
-                for (int i = 0; i < sizeOfItems; i++) {
-                    auto item = ITEM_MANAGER->baseItemArrayList.at(i);
-                    itemIds[i] = (*item)->getItKind();
-                    itemVariants[i] = (*item)->getItVariation();
+                else
+                {
+                    ftManager *fighter = FIGHTER_MANAGER;
+                    const GameGlobal *game = GAME_GLOBAL;
+                    int sizeOfItems = ITEM_MANAGER->baseItemArrayList.size();
+
+                    auto* replay = new Replay();
+                    replay->numItems = sizeOfItems;
+                    replay->frameCounter = game->gameFrame->frameCounter;
+                    replay->persistentFrameCounter = game->gameFrame->persistentFrameCounter;
+
+                    if(sizeOfItems > 0) {
+                        for (int i = 0; i < sizeOfItems; i++) {
+                            auto item = ITEM_MANAGER->baseItemArrayList.at(i);
+                            replay->items[i].itemId = (*item)->getItKind();
+                            replay->items[i].itemVariant = (*item)->getItVariation();
+                        }
+                    }
+                    replay->numPlayers = fighter->getEntryCount();
+                    for(int i = 0; i < fighter->getEntryCount(); i++)
+                    {
+                        const soStatusModuleImpl *stageObject = fighter->getFighter(fighter->getEntryIdFromIndex(i))->modules->statusModule;
+                        replay->players[i].actionState = stageObject->action;
+                        replay->players[i].inputs.attack = fighter->getInput(fighter->getEntryIdFromIndex(i))->buttons.attack;
+                        replay->players[i].inputs.special = fighter->getInput(fighter->getEntryIdFromIndex(i))->buttons.special;
```

Comment:
@JaredWhiteOne let's make this a little more performant by accessing each object just once instead of on each assignment, it also helps reading this better.
i.e:
```cpp
auto fighter* = fighterManager->getFighter(fighter->getEntryIdFromIndex(i));
auto player = replay->players[i];
auto fighterInput = fighterManager->getInput(fighter);
const soStatusModuleImpl *stageObject = fighter->modules->statusModule;


player.actionState = stageObject->action;
player.inputs.attack = fighterInput->buttons.attack;
// ...
player.inputs.jump = fighterInput->buttons.jump;
```

Reply:
This depends on https://github.com/Brawlback-Team/brawlback-asm/pull/17/files#r874897672 btw.